.github/workflows/lerna-publish.yml

```diff
@@ -23,7 +25,47 @@ jobs:
           registry-url: https://wombat-dressing-room.appspot.com
 
       - run: npm install --no-save --no-package-lock --legacy-peer-deps --registry=https://registry.npmjs.org/
+      - run: npm run build
+
+      # setup test env so prepublishOnly lifecycle script can run
+      - name: Mock looker.ini
+        run: |
+          echo "[Looker]" > looker.ini
+          echo "base_url=https://self-signed.looker.com:19999" >> looker.ini
+          echo "client_id=id" >> looker.ini
+          echo "client_secret=secret" >> looker.ini
+          echo "verify_ssl=False" >> looker.ini
+          echo "timeout=31" >> looker.ini
+          echo "[Looker31]" >> looker.ini
+          echo "base_url=https://self-signed.looker.com:19999" >> looker.ini
+          echo "verify_ssl=False" >> looker.ini
+          echo "timeout=30" >> looker.ini
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0.2.0
+        with:
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          service_account_key: ${{ secrets.GCP_AR_READER_SA_KEY }}
+          export_default_credentials: true
+
+      - name: Authenticate Artifact Repository
+        run: gcloud auth configure-docker us-west1-docker.pkg.dev --quiet
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+
+      - name: Pull and run Looker docker image
+        # TODO: can we cache some layers of the image for faster download?
+        # we probably don't want to cache the final image for IP security...
+        run: |
+          docker pull --quiet us-west1-docker.pkg.dev/cloud-looker-sdk-codegen-cicd/looker/21_4
+          # set $LOOKER_OPTS to --no-ssl if we want to turn off ssl
```

Comment:
FWIW if we've got ssl always working I don't think we need to document this. But it certainly doesn't hurt anything.

Reply:
shoot, all `methods.spec.ts` tests failing like this:
```
● LookerNodeSDK › sudo › login/logout

    POST /api/3.1/login <!DOCTYPE html>
    <html>
    <head>
        <title>Looker Unavailable (502)</title>
        <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,600' rel='stylesheet' type='text/css'>
        <link rel="icon" type="image/x-icon"  href="data:image/png;base64,AAABAAIAEBAAAAEAIABoBAAAJgAAACAgAAABACAAqBAAAI4EAAAoAAAAEAAAACAAAAABACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAO+HQALzhEFO84RBvvOEQfLzhEHy84RBvPOEQUwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAO6HQALzhEGG9IVB//OEQu7zhEKu84RBrvOFQfDzhEH/84RBggAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADzhEFM84VB//SFQbjzhEESAAAAAAAAAADzhUES84VBuvOEQf/zhEFGAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA84RBvPOEQvDzhEESAAAAAAAAAAAAAAAAAAAAAPOFQRTzhEHy84RBtgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPOEQe7zhEGyAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA84RBtvOEQuoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADzhEHu84RBtAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPOEQbb0hEHoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA84RBtvOEQfLzhEEWAAAAAAAAAAAAAAAAAAAAAPOEQRjzhEH084RBsgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPOFQUTzhUH/84RBwvOEQRYAAAAAAAAAAPOEQRrzhEHE84VB//OEQUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA84VBePOEQf/zhEE484RBsPOEQbzzhEH284RB//OEQnQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADVC6SCgaIta84RBPPOEQuTzhELk84RBrvOEQT7ugEACAAAAAAAAAAAAAA...
        <style type="text/css">
          body ***
            background-color: #2e343f;
            color: white;
            height: auto;
            font-family: Open Sans, Helvetica, Arial, sans-serif;
          ***
          .message ***
            width: 100%;
            max-width: 760px;
            margin: 0 auto;
            margin-top: 135px;
            text-align: center;
          ***
          h2, h3 ***
            font-weight: normal;
          ***
          a ***
            color: white;
          ***
        </style>
    </head>
    <body>

      <div class="message">

        <img width="210" height="84" alt="Looker" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAaQAAACoCAYAAACv+Zr6AAAACXBIWXMAAAsTAAALEwEAmpwYAAA4KmlUWHRYTUw6Y29tLmFkb2JlLnhtcAAAAAAAPD94cGFja2V0IGJlZ2luPSLvu78iIGlkPSJXNU0wTXBDZWhpSHpyZVN6TlRjemtjOWQiPz4KPHg6eG1wbWV0YSB4bWxuczp4PSJhZG9iZTpuczptZXRhLyIgeDp4bXB0az0iQWRvYmUgWE1QIENvcmUgNS42LWMxMzIgNzkuMTU5Mjg0LCAyMDE2LzA0LzE5LTEzOjEzOjQwICAgICAgICAiPgogICA8cmRmOlJERiB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiPgogICAgICA8cmRmOkRlc2NyaXB0aW9uIHJkZjphYm91dD0iIgogICAgICAgICAgICB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iCiAgICAgICAgICAgIHhtbG5zOmRjPSJodHRwOi8vcHVybC5vcmcvZGMvZWxlbWVudHMvMS4xLyIKICAgICAgICAgICAgeG1sbnM6cGhvdG9zaG9wPSJodHRwOi8vbnMuYWRvYmUuY29tL3Bob3Rvc2hvcC8xLjAvIgogICAgICAgICAgICB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIKICAgICAgICAgICAgeG1sbnM6c3RFdnQ9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZUV2ZW50IyIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzE...

        <h1>Looker is unavailable.</h1>

        <h2>If you typed in a URL, double-check the spelling.</h2>
          <h2>This may also be due to a temporary condition such as an outage, <a href="https://docs.looker.com/relnotes/hosted-maintenance-hours">scheduled maintenance</a> or upgrade.</h2>
        <br>
        <h3>
            If this message persists or you have any concerns, <br> contact us
            from <a href="https://help.looker.com">help.looker.com</a> and we'll respond promptly.
        </h3>

      </div>

    </body>
    </html>

      200 |
      201 |     if ('message' in _error) ***
    > 202 |       return new Error(response.error.message.toString());
          |              ^
      203 |     ***
      204 |   ***
      205 |

      at sdkError (packages/sdk-rtl/lib/transport.js:202:14)
      at packages/sdk-rtl/lib/transport.js:225:13
      at asyncGeneratorStep (packages/sdk-rtl/lib/transport.js:19:103)
      at _next (packages/sdk-rtl/lib/transport.js:21:194)
```

Reply:
this is the env:

```
Run $(npm bin)/lerna publish from-package --yes --no-verify-access
  $(npm bin)/lerna publish from-package --yes --no-verify-access
  shell: /usr/bin/bash -e ***0***
  env:
    LOOKERSDK_CLIENT_ID: ***
    LOOKERSDK_CLIENT_SECRET: ***
    NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc
    NODE_AUTH_TOKEN: ***
    CLOUDSDK_METRICS_ENVIRONMENT: github-actions-setup-gcloud
    GCLOUD_PROJECT: ***
    GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/sdk-codegen/sdk-codegen/55f8ca5b-474f-4367-bdc0-f2c92df49517
    pythonLocation: /opt/hostedtoolcache/Python/3.9.2/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.2/x64/lib
```

Reply:
ah, missing these from the tssdk-ci.yml workflow:
```
  LOOKERSDK_BASE_URL: https://localhost:20000
  LOOKERSDK_VERIFY_SSL: false
```

but not sure how I'm even getting a response at all? how does the github runner possibly have access to an instance at `https://self-signed.looker.com:19999` ?

Reply:
whoohoo, that fixed it!